src/search.js

```diff
@@ -25,37 +25,38 @@ var Search = React.createClass({
   mixins: [Router.State, Router.Navigation],
 
   getInitialState() {
-    if (!this.props.data.searchResults) {
-      if (typeof window !== "undefined") {
-        const q = new URLSearchParams(location.search).get("q");
-        if (q) {
-          window.location = `/search/${q}`;
-          return;
-        }
-      }
-
-      this.transitionTo("home");
+    const q = new URLSearchParams(location.search).get("q");
+    if (!this.props.data || !this.props.data.searchResults) {
+      if (q) {
+      window.location = `/search/${encodeURIComponent(q)}`;
```

Comment:
Do me a favor and enable prettier to run on save in your IDE. I'm seeing whitespace problems in GitHub.

Reply:
I've done this and merged it.